_overviews/scala3-book/collections-classes.md

```diff
@@ -0,0 +1,600 @@
+
+```scala
+for ((k,v) <- states) println(s"key: $k, value: $v")
+```
+
+The REPL shows how this works:
+
+````
+scala> for ((k,v) <- states) println(s"key: $k, value: $v")
+key: AK, value: Alaska
+key: AL, value: Alabama
+key: AZ, value: Arizona
+````
+
+### Accessing Map elements
+
+Access map elements by specifying the desired key value in parentheses:
+
+```scala
+val ak = states("AK")   // ak: String = Alaska
+val al = states("AL")   // al: String = Alabama
+```
+
+In practice you’ll also use methods like `keys`, `keySet`, `keysIterator`, `for` loops, and higher-order functions like `map` to work with `Map` keys and values.
+
+### Adding elements to a Map
+
+Add elements to an immutable map using `+` and `++`, remembering to assign the result to a new variable:
+
+```scala
+val a = Map(1 -> "one")    // a: Map(1 -> one)
+val b = a + (2 -> "two")   // b: Map(1 -> one, 2 -> two)
+val c = b + (
+  3 -> "three",
+  4 -> "four"
+)
+// c: Map(1 -> one, 2 -> two, 3 -> three, 4 -> four)
+```
+
+## Removing elements from a Map
+
+Remove elements from an immutable map using `-` or `--` and the key values to remove, remembering to assign the result to a new variable:
+
+```scala
+val a = Map(
+  1 -> "one",
+  2 -> "two",
+  3 -> "three",
+  4 -> "four"
+)
+
+a - 4         // Map(1 -> one, 2 -> two, 3 -> three)
+a - 4 - 3     // Map(1 -> one, 2 -> two)
+```
+
+## Updating Map elements
+
+To update elements in an immutable map, use the `updated` method while assigning the result to a new variable:
+
+```scala
+val a = Map(
+  1 -> "one",
+  2 -> "two",
+  3 -> "three"
+)
+
+val b = a.updated(3, "THREE!")   // Map(1 -> one, 2 -> two, 3 -> THREE!)
+```
+
+## Traversing a Map
+
+As shown earlier, this is a common way to manually traverse elements in a map using a `for` loop:
+
+```scala
+val states = Map(
+  "AK" -> "Alaska",
+  "AL" -> "Alabama",
+  "AZ" -> "Arizona"
+)
+
+for ((k,v) <- states) println(s"key: $k, value: $v")
+```
+
+That being said, there are _many_ ways to work with the keys and values in a map. Common `Map` methods include `foreach`, `map`, `keys`, and `values`. 
+
+Scala has many more specialized `Map` classes available, including `CollisionProofHashMap`, `HashMap`, `LinkedHashMap`, `ListMap`, `SortedMap`, `TreeMap`, `WeakHashMap`, and more. See the Reference documentation for more details on `Map` methods and these specialized subclasses.
+
+
+
+## Working with Sets
+
+{% comment %}
+Types of Set classes:
+- BitSet, HashSet, LinkedHashSet, ListSet, TreeSet, SortedSet
+{% endcomment %}
+
+The [Scala Set class]({{site.baseurl}}/overviews/collections-2.13/sets.html) is an iterable collection with no duplicate elements.
+
+Scala has both mutable and immutable `Set` classes. This section demonstrates the _mutable_ `Set` class.
+
+### Adding elements to a Set
+
+The mutable `Set` class isn’t in scope by default, so first you need to import it:
+
+```scala
+val set = scala.collection.mutable.Set[Int]()
```

Comment:
```suggestion
import scala.collection.mutable.Set
val set = Set.empty[Int]
```

Reply:
I would suggest the following, overloading `Set` can be confusing.
```suggestion
import scala.collection.mutable
val set = mutable.Set[Int]()
```